addon/ActivityStreams.js

```diff
@@ -95,6 +95,10 @@ function ActivityStreams(metadataStore, tabTracker, telemetrySender, options = {
     fetchNewMetadata: (links, type) => {
       const event = this._tabTracker.generateEvent({source: type});
       return this._previewProvider.asyncSaveLinks(links, event);
+    },
+    fetchNewMetadataLocally: (links, type) => {
+      const event = this._tabTracker.generateEvent({source: type});
+      return this._pageScraper.asyncFetchLinks(links, event);
```

Comment:
I can't actually do this - the feeds expect a function to be returned for getData to work properly. If I do this I get an error in the console saying: ```this.options.fetchNewMetadataLocally(...) is undefined```